packages/material-ui-lab/src/MasonryItem/MasonryItem.js

```diff
@@ -0,0 +1,158 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import clsx from 'clsx';
+import {
+  createUnarySpacing,
+  getValue,
+  handleBreakpoints,
+  unstable_resolveBreakpointValues as resolveBreakpointValues,
+} from '@material-ui/system';
+import { unstable_useForkRef as useForkRef } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import { styled, useThemeProps } from '@material-ui/core/styles';
+import { getMasonryItemUtilityClass } from './masonryItemClasses';
+import MasonryContext from '../Masonry/MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryItemUtilityClass, classes);
+};
+
+export const style = ({ styleProps, theme }) => {
+  let styles = {
+    width: '100%',
+    '& > *': {
+      // all contents should have a width of 100%
+      width: '100%',
+    },
+    visibility: styleProps.height ? 'visible' : 'hidden',
+    gridColumnEnd: `span ${styleProps.columnSpan}`,
+  };
+
+  const base = Object.keys(theme.breakpoints.values).reduce((acc, breakpoint) => {
+    if (styleProps.spacing[breakpoint] != null) {
+      acc[breakpoint] = true;
+    }
+    return acc;
+  }, {});
```

Comment:
I understand that there exists a couple of `.reduce` in the codebase using a similar pattern. But this pattern is needlessly complicated. We don't reduce anything here. This really should be just a `forEach`:

```suggestion
  const base = {};
  const base = Object.keys(theme.breakpoints.values).forEach((breakpoint) => {
    if (styleProps.spacing[breakpoint] != null) {
      acc[breakpoint] = true;
    }
  });
```

Saw a couple of other introductions of `reduce` that aren't actually reducing a list.